### PR TITLE
Periodic sync separate screen.

### LINKED ISSUE #2694: Refactor the FHIR Engine demo app to showcase the APIs
**Is your feature request related to a problem? Please describe.**
At the moment the demo app is not organised by fhir engine APIs - to make it easier for developers to find api usage examples, we should make fhir engine demo app more modularised towards showcasing individual apis.

**Describe the solution you'd like**
Proposing the following changes:

- [ ] combine patient list and search pages - at the moment they are linked to the same screen
- [ ] remove register a new patient page - this should be in the sdc catalog app
- [ ] enhance search api demo - showcasing filters, reverse includes etc. to make this easier, we may want to use hardcoded patient data
- [ ] add demo for data access apis - create, read, update, delete

**Describe alternatives you've considered**
na

**Additional context**
na

**Would you like to work on the issue?**
@santosh-pingle

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2694 

**Description**

- Periodic sync as separate screen.
- 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature 
**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
https://github.com/user-attachments/assets/268e8f07-52f1-42ee-ab35-bb47c03c338b

File: demo/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -70,5 +70,6 @@
   <string name="sync">Sync</string>
   <string name="last_sync_timestamp">Last sync: %1$s</string>
   <string name="current_status">Sync status: %1$s</string>
-  <string name="manual_sync">Sync</string>
+  <string name="last_sync_status">Last sync status: %1$s</string>
+  <string name="periodic_sync">Periodic Sync</string>
```
Comment (jingtang10) [inline demo/src/main/res/values/strings.xml:74]:
```suggestion
  <string name="periodic_sync">Periodic sync</string>
```

and rename `Sync` on line 70 to `One-time sync` to contrast this.

File: demo/src/main/res/layout/fragment_home.xml
Comment (jingtang10) [inline demo/src/main/res/layout/fragment_home.xml:1]: in the screenshot, the text in different items are not aligned - there's an indentation possibly caused by the size of the icon - pls adjust.

File: demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncFragment.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?,
+  ): View {
+    return inflater.inflate(R.layout.periodic_sync, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
Comment (jingtang10) [inline demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncFragment.kt:47]: i think the last sync statuses can be shown from the begining.

**Review by jingtang10 — CHANGES_REQUESTED** [review]
thanks @santosh-pingle!

a few things:

1. in the same pr we can remove sync from patient list and search screen
2. fix the failed sync in the video - is it a transient problem in your testing, or is there something wrong with the sync?
3. i think a progress bar would probably be really useful in the sync screens.

i love the simplicity of the screen - it really is a much clearer demo.

File: demo/src/main/res/values/strings.xml (hunk 1 above)
Comment (santosh-pingle) [reply demo/src/main/res/values/strings.xml]: done.

File: demo/src/main/res/layout/fragment_home.xml
Comment (santosh-pingle) [reply demo/src/main/res/layout/fragment_home.xml:1]: done.

File: demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncFragment.kt (hunk 2 above)
Comment (santosh-pingle) [reply demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncFragment.kt:46]: Done. Except when the sync is happening for the first time, in which case there won't be a last sync state.

**Comment by santosh-pingle** [comment]
https://github.com/user-attachments/assets/f8eeaca0-24ba-42d9-8740-b6ac8cabc538

File: demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  val lastSyncStatusFlow: StateFlow<String?> = _lastSyncStatusFlow
+
+  private val _lastSyncTimeFlow = MutableStateFlow<String?>(null)
+  val lastSyncTimeFlow: StateFlow<String?> = _lastSyncTimeFlow
+
+  private val _currentSyncStatusFlow = MutableStateFlow<String?>(null)
+  val currentSyncStatusFlow: StateFlow<String?> = _currentSyncStatusFlow
+
+  private val _progressFlow = MutableStateFlow<Int?>(null)
+  val progressFlow: StateFlow<Int?> = _progressFlow
```
[the remark this answers is not in the document]
Comment (santosh-pingle) [reply demo/src/main/java/com/google/android/fhir/demo/PeriodicSyncViewModel.kt]: done.

**Comment by santosh-pingle** [comment]
https://github.com/user-attachments/assets/03767c6a-39d5-4e17-86c7-ac8bcd42155f

**Review by jingtang10 — APPROVED** [review]
thanks santosh!

**Pull request merged**